### PR TITLE
Ghost bar

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Essentially implements #20854 (vast majority of reasoning is stated in said doc)
Respawnability moved to a trait instead of list of mobs, ghost notifications are sent to mobs with this respawnable trait

TODO (In order of difficulty):

- [ ] Maybe delete the mob when they spawn in as another ghost thing
- [ ] Implement the nebula nook map
- [ ] Read only intercom + newscaster
- [ ] Deadchat intercom
- [ ] Do some genuinely vile shit with planemasters to get it to function like a TV

## Why It's Good For The Game
This is a draft but essentially, observing just doesn't have a lot going on, it sucks so let's let them chill out and have some fun.
Dwarves will be real eventually. 

## Testing
Draft
## Changelog
:cl:
add: The ghost bar
/:cl:

### CODE REVIEW

File: code/__HELPERS/traits.dm [hunk 1]
```diff
@@ -202,6 +202,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_UNREVIVABLE 		"unrevivable" // Prevents changeling revival
 #define TRAIT_CULT_IMMUNITY		"cult_immunity"
 #define TRAIT_FLATTENED			"flattened"
+#define TRAIT_RESPAWNABLE		"respawnable" // Allows respawnable people to
```
Comment (lewcc) [inline code/__HELPERS/traits.dm:205]: Honestly, I feel like you might want to make this change in how respawnability works its own atomic PR. It's a big thing on its own.

**Comment by Christasmurf** [comment]
I assume this has already been thought of but just in case, I must ask
1. Will this use a long range cloner/other similar means to make the ghost into a person to mess around in the bar?
Or will it
2. Spawn a completely new mob and put the person in control of it, with magic mirrors available for them to be whichever character they want?

The reason I ask is because if there are traitors who need to assassinate someone, for example, and that person then respawns in the bar, the traitor will fail that objective despite having killed the person and them never getting revived; this is if you go with the first option without some major reworking of systems. It's the main reason we admins rarely give ghosts an opportunity to respawn like this. As this is something I would assume would be in the CC zlevel, you can't have it zlevel locked to ignore them either as it will still apply to the pods and emergency shuttle in transit.

Just some food for thought.

**Comment by SteelSlayer** [comment]
Haven't seen any activity on this PR since it's opening, so it's being marked stale. This PR will be closed after a week if no progress is made.

**Comment by GDNgit** [comment]
Will be finished probably at the start of September, will reopen then

**Pull request closed without merging**